src/main/java/com/metamx/common/CompressionUtils.java

```diff
@@ -0,0 +1,372 @@
+      }
+    }
+  }
+
+  public static FileUtils.FileCopyResult unzip(
+      final ByteSource byteSource,
+      final File outDir,
+      boolean cacheLocally
+  ) throws IOException
+  {
+    return unzip(byteSource, outDir, Predicates.<Throwable>alwaysTrue(), cacheLocally);
+  }
+
+  public static FileUtils.FileCopyResult unzip(final File pulledFile, final File outDir) throws IOException
+  {
+    if (!(outDir.exists() && outDir.isDirectory())) {
+      throw new ISE("outDir[%s] must exist and be a directory", outDir);
+    }
+    log.info("Unzipping file[%s] to [%s]", pulledFile, outDir);
+    final ArrayList<FileUtils.FileCopyResult> children = new ArrayList<>();
+    try (final ZipFile zipFile = new ZipFile(pulledFile)) {
+      final Enumeration<? extends ZipEntry> enumeration = zipFile.entries();
+      while (enumeration.hasMoreElements()) {
+        final ZipEntry entry = enumeration.nextElement();
+        children.add(
+            FileUtils.retryCopy(
+                new ByteSource()
+                {
+                  @Override
+                  public InputStream openStream() throws IOException
+                  {
+                    return new BufferedInputStream(zipFile.getInputStream(entry));
+                  }
+                },
+                Paths.get(outDir.getAbsolutePath(), entry.getName()).toFile(),
+                FileUtils.MD5ERROR,
+                DEFAULT_RETRY_COUNT
+            )
+        );
+      }
+    }
+    return new FileUtils.FileCopyResult(outDir.toPath(), children);
+  }
+
+  public static FileUtils.FileCopyResult unzip(InputStream in, File outDir) throws IOException
+  {
+    try (final ZipInputStream zipIn = new ZipInputStream(in)) {
+      final ArrayList<FileUtils.FileCopyResult> children = new ArrayList<>();
+      ZipEntry entry;
+      while ((entry = zipIn.getNextEntry()) != null) {
+        final File file = new File(outDir, entry.getName());
+        MessageDigest digest = StreamUtils.copyToFileWithInputDigest(zipIn, file);
+        children.add(new FileUtils.FileCopyResult(file.toPath(), digest));
+        zipIn.closeEntry();
+      }
+      return new FileUtils.FileCopyResult(outDir.toPath(), children);
+    }
+  }
+
+  /**
+   * gunzip the file to the output. If out is a directory, it will gunzip the pulled file using the name less `.gz`.
+   * If out is a not a directory, it will simply gunzip to that file
+   *
+   * @param pulledFile The source of the gz data
+   * @param out        Either a directory or a target file
+   *
+   * @return The result of the file copy
+   *
+   * @throws IOException
+   */
+  public static FileUtils.FileCopyResult gunzip(final File pulledFile, File out) throws IOException
+  {
+    if (out.isDirectory()) {
+      final String name = pulledFile.getName();
+      if (!isGz(name)) {
+        throw new IAE("[%s] is not a .gz file", pulledFile.getAbsolutePath());
+      }
+      final File outFile = Paths.get(out.getAbsolutePath(), getGzBaseName(name))
```

Comment:
most command like tools typically report errors in terms of relative paths, not absolute paths, unless necessary. Absolute paths can become terribly confusing because you have to decide whether to resolve symbolic links or not, which is something better left to the user.